src/chapters/sql-part1/studio.rst

```diff
@@ -0,0 +1,269 @@
+
+.. sourcecode:: sql
+   :linenos:
+
+   CREATE TABLE movies (
+      movie_id INTEGER PRIMARY KEY AUTO_INCREMENT,
+      title VARCHAR(120),
+      year INTEGER,
+      director VARCHAR(80)
+   );
+
+(Take a moment to discuss with a partner why the two ``VARCHAR`` parameters
+differ).
+
+Directors Table
+^^^^^^^^^^^^^^^
+
+Next, let's consider the properties we'd want for a ``director`` object in
+Java. These could include a ``first`` name, a ``last`` name, and maybe the
+``country`` where the director was born.
+
+Once we identify the data types for these properties, we can write the SQL
+command to create the ``directors`` table. Also, we should include a
+``director_id`` column in order to create a valid primary key.
+
+.. _directors-table:
+
+.. sourcecode:: sql
+   :linenos:
+
+   CREATE TABLE directors (
+      director_id INTEGER PRIMARY KEY AUTO_INCREMENT,
+      first VARCHAR(40),
+      last VARCHAR(40),
+      country VARCHAR(80)
+   );
+
+Relating the Tables
+^^^^^^^^^^^^^^^^^^^^
+
+Before we go further, let's take a moment to think about how these two tables
+relate to each other, and whether we may want to modify the column ``director``
+in the ``movies`` table. Understanding the common *one-to-many* relationship
+that exists between database tables helps us set up the best design for
+``movies`` and ``directors``. A director can work on multiple movies, but each
+movie may only have one director.
+
+.. admonition:: Note
+
+   It may be useful to review `this article <http://www.databaseprimer.com/pages/relationship_1tox/>`__
+   to refresh your memory on the one-to-many concept.
+
+In order to reflect a one-to-many relationship in the table definitions, let's
+modify the ``director`` column in ``movies`` to become ``director_id``. It
+should therefore be an ``INTEGER`` instead of the ``VARCHAR`` we used earlier
+when we thought the ``director`` column would hold the name of a person.
+
+We also will want to make the ``director_id`` column in ``movies`` a
+*foreign key*, so that it links directly to the ``director_id`` column in the
+``directors`` table. Go ahead and write the SQL to drop the old ``movies``
+table, and then write the SQL command to create a new table with these changes.
+
+.. sourcecode:: sql
+   :linenos:
+
+   DROP TABLE movies;
+
+   CREATE TABLE movies (
+      movie_id INTEGER PRIMARY KEY AUTO_INCREMENT,
+      title VARCHAR(120),
+      year INTEGER,
+      director_id INTEGER,
+      FOREIGN KEY (director_id) REFERENCES directors(director_id)
+   );
+
+.. admonition:: Note
+
+   If needed, here is a set of review articles:
+
+   #. `SQL Drop Table <https://www.w3schools.com/sql/sql_drop_table.asp>`__,
+   #. `SQL Create Table <https://www.w3schools.com/sql/sql_create_table.asp>`__,
+   #. `SQL Foreign Key <https://www.w3schools.com/sql/sql_foreignkey.asp>`__.
+
+   Follow the MySQL syntax instructions when there is a syntax variation
+   between the major databases.
```

Comment:
They haven't covered ``DROP TABLE`` either.

Reply:
Unfortunately, this step is necessary for the walkthrough, and I think that it is straightforward enough not to present a problem.